can/interfaces/slcan.py

```diff
@@ -158,18 +158,14 @@ def _read(self, timeout: Optional[float]) -> Optional[str]:
                 new_byte = self.serialPortOrig.read(1)
                 if new_byte:
                     self._buffer.extend(new_byte)
-                else:
-                    if _timeout.expired():
-                        break
-                    else:
-                        continue
-
-                for terminator in (self._ERROR, self._OK):
-                    if terminator in self._buffer:
-                        i = self._buffer.index(terminator) + 1
+                    if new_byte in (self._ERROR, self._OK):
+                        i = self._buffer.index(new_byte) + 1
                         string = self._buffer[:i].decode()
                         del self._buffer[:i]
                         return string
+                else:
+                    if _timeout.expired():
```

Comment:
Do we really need `_timeout` to handle timeout?

The timeout is determined when `pyserial.read` returns `b''`.

Reply:
The timeout of `pyserial.read` is reset on every `read` call, so it might add up. The timeout argument of the user should be respected, `_timeout` takes care of that.

Reply:
I think the `timeout's add up` should be ignored. 
For python on windows, the timer's resolution is about 16ms, obviously it is not very accurate.
And I think the `timeout add up` is not as big as the impact of timer jitter.

Reply:
That depends on the `timeout` value, It might be seconds, or minutes, or hours, i don't know what users might be doing.